BitwardenShared/Core/Vault/Repositories/VaultRepositoryTests.swift

```diff
@@ -1087,6 +1087,59 @@ class VaultRepositoryTests: BitwardenTestCase { // swiftlint:disable:this type_b
         XCTAssertTrue(isDisabled)
     }
 
+    /// `getItemTypesUserCanCreate()` gets the user's available item types for item creation
+    /// when feature flag is true and true are policies enabled.
+    @MainActor
+    func test_getItemTypesUserCanCreate() async throws {
+        stateService.activeAccount = .fixture()
+        configService.featureFlagsBool[.removeCardPolicy] = true
+        policyService.policyAppliesToUserPolicies = [
+            .fixture(
+                enabled: true,
+                id: "restrict_item_type",
+                organizationId: "org1",
+                type: .restrictItemTypes,
+            ),
+        ]
+
+        let result = await subject.getItemTypesUserCanCreate()
+        XCTAssertEqual(
+            result,
+            CipherType.canCreateCases.filter { $0 != .card }.reversed(),
+        )
```

Comment:
🎨 IMO tests should not use the same logic as what's testing as it could be a bit hard to catch bugs on the implementation given that it would use the same as in the test method. Basically this would test the logic against itself instead of a known-good result.
I believe the expected result in this case would need to be the explicit expected enum cases ordered in an array: `[.sshKey, .secureNote, .identity, .login]`.
Same applies to the other tests for `getItemTypesUserCanCreate`.